tests/integration/test_logcollector/test_macos/test_macos_file_status_predicate.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (C) 2015-2021, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+import os
+import pytest
+import re
+
+from wazuh_testing.logcollector import DEFAULT_AUTHD_REMOTED_SIMULATOR_CONFIGURATION
+from wazuh_testing.tools.configuration import load_wazuh_configurations
+from wazuh_testing.tools import LOG_FILE_PATH, WAZUH_PATH
+from wazuh_testing.tools.monitoring import FileMonitor
+from wazuh_testing.tools.monitoring import wait_file
+from wazuh_testing.tools.file import read_json
+
+# Marks
+pytestmark = [pytest.mark.darwin, pytest.mark.tier(level=0)]
+
+# Configuration
+test_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+configurations_path = os.path.join(test_data_path, 'wazuh_macos_file_status_predicate.yaml')
+
+parameters = [{'ONLY_FUTURE_EVENTS': 'yes'}, {'ONLY_FUTURE_EVENTS': 'no'}]
+metadata = [{'only-future-events': 'yes'}, {'only-future-events': 'no'}]
+
+# Configuration data
+configurations = load_wazuh_configurations(configurations_path, __name__, params=parameters, metadata=metadata)
+configuration_ids = [f'{x["ONLY_FUTURE_EVENTS"]}' for x in parameters]
+
+file_status_path = os.path.join(WAZUH_PATH, 'queue', 'logcollector', 'file_status.json')
+
+wazuh_log_monitor = FileMonitor(LOG_FILE_PATH)
+
+# Maximum waiting time in seconds to find the logs on ossec.log
+wazuh_log_monitor_timeout = 30
+
+# Time in seconds to update the file_status.json
+file_status_update_time = 4
+
+local_internal_options = {'logcollector.vcheck_files': str(file_status_update_time)}
+
+
+@pytest.fixture(scope='module')
+def get_local_internal_options():
+    """Get configurations from the module."""
+    return local_internal_options
+
+
+def extra_configuration_before_yield():
+    """Delete file status file."""
+    os.remove(file_status_path) if os.path.exists(file_status_path) else None
+
+
+def callback_log_bad_predicate(line):
+    match = re.match(r'.*Execution error \'log:', line)
+    if match:
+        return True
+    return None
+
```

Comment:
This should be defined following callback function structure in `logcollector` module.

Reply:
This could do the work:

```
def callback_log_bad_predicate():
    """Create a callback to detect logcollector bad predicate error"""
    msg = "Execution error \'log:'"
    return monitoring.make_callback(pattern=msg, prefix=prefix, escape=True)
```